supervisor.go

```diff
@@ -41,9 +42,37 @@ func NewSupervisor(configFile string) *Supervisor {
 		configFile: configFile,
 		procMgr:    process.NewManager(),
 		config:     new(config.Config),
+		done:       make(chan struct{}),
 	}
 }
 
+// Close closes the server, stopping all processes gracefully.
+func (s *Supervisor) Close() error {
+	// Updating with the empty configuration should gracefully stop all processes
+	// in the correct order and stop and tear down any current network
+	// requests.
+	if err := s.procMgr.Update(context.TODO(), &config.Config{}); err != nil {
+		zap.L().Info("cannot terminate processes", zap.Error(err))
+	}
+	s.mu.Lock()
```

Comment:
defer unlock

Reply:
Where is this deferred?

Reply:
it's not! it was me in a concert with minimal connectivity perusing the code and commenting to remind myself that we must unlock the mutext here.

Reply:
Fixed